src/repart/repart.c

```diff
@@ -7878,6 +7917,11 @@ static int help(void) {
                "                          Specify how to interpret the certificate from\n"
                "                          --certificate=. Allows the certificate to be loaded\n"
                "                          from an OpenSSL provider\n"
+               "     --append-signature=HASH:SIG\n"
+               "                          Specify root hash and pkcs7 signature of root hash for\n"
+               "                          verity as a tuple of hex encoded hash and a DER\n"
+               "                          encoded PKCS7, either as a path to a file or as an\n"
```

Comment:
All of our new tools take PEM encoded crypto things, I'm pretty sure this one should as well unless there's a good argument not to.

Reply:
Let's call this `--join-signature` maybe

Reply:
the kernel expects DER encoded pkcs7, it cannot deal with PEM and rejects it

Reply:
That's why we convert it to DER in repart

Reply:
> Let's call this `--join-signature` maybe

I think I prefer append, that sounds like it would join signatures together

Reply:
> That's why we convert it to DER in repart (assuming that's possible)

That's a libssl can of worms I'd really rather not open unless there's a very real use case for it. It can always be added later if there's a demand for it?

Reply:
> I think I prefer append, that sounds like it would join signatures together

We have --join-profile, --join-pcrsig in ukify already, naming this one --append-signature would look really out of place

Reply:
fine